internal/blockdb/tui/presenter/highlight.go

```diff
@@ -0,0 +1,43 @@
+package presenter
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type Highlight struct {
+	rx *regexp.Regexp
+}
+
+// NewHighlight returns a presenter that adds regions around text that matches searchTerm.
+func NewHighlight(searchTerm string) *Highlight {
+	searchTerm = strings.TrimSpace(searchTerm)
+	if searchTerm == "" {
+		return &Highlight{}
+	}
+	// Should always be valid with regexp.QuoteMeta.
+	return &Highlight{rx: regexp.MustCompile(fmt.Sprintf(`(?is)(%s)`, regexp.QuoteMeta(searchTerm)))}
```

Comment:
I suppose the `s` flag, which is intended to "let `.` match `\n`", will not do anything here because we will never have a `.` after the QuoteMeta.

Reply:
I'll try it without that. My intention was to match multi-line.

Reply:
Works fine without the flag. Nice catch.